src/Tests/Microsoft.NET.Build.Tests/MetadataUpdaterSupportTest.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class MetadataUpdaterSupportTest : SdkTest
+    {
+        public MetadataUpdaterSupportTest(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [CoreMSBuildOnlyFact] // Running on desktop causes failures attempting to restore M.NETCore.App.WinHost.
```

Comment:
That seems like a bug somewhere.... This is a super simple test....

cc @dsplaisted

Reply:
I was misled by the error messages. It looked like it was restore failures, but it looks like it's because of the new compiler feature that needs a newer VS on the Helix machine:

```
"C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\HelloWorld.csproj" (Build target) (1:7) ->
        (CoreCompile target) -> 
          C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\obj\Debug\net6.0\HelloWorld.ImplicitNamespaceImports.cs(2,1): error CS8652: The feature 'global using directive' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version. [C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\HelloWorld.csproj]
          C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\obj\Debug\net6.0\HelloWorld.ImplicitNamespaceImports.cs(3,1): error CS8652: The feature 'global using directive' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version. [C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\HelloWorld.csproj]
          C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\obj\Debug\net6.0\HelloWorld.ImplicitNamespaceImports.cs(4,1): error CS8652: The feature 'global using directive' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version. [C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\HelloWorld.csproj]
          C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\obj\Debug\net6.0\HelloWorld.ImplicitNamespaceImports.cs(5,1): error CS8652: The feature 'global using directive' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version. [C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\HelloWorld.csproj]
          C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\obj\Debug\net6.0\HelloWorld.ImplicitNamespaceImports.cs(6,1): error CS8652: The feature 'global using directive' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version. [C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\HelloWorld.csproj]
          C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\obj\Debug\net6.0\HelloWorld.ImplicitNamespaceImports.cs(7,1): error CS8652: The feature 'global using directive' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version. [C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\HelloWorld.csproj]
          C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\obj\Debug\net6.0\HelloWorld.ImplicitNamespaceImports.cs(8,1): error CS8652: The feature 'global using directive' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version. [C:\h\w\B468094B\t\dotnetSdkTests\dft2qt2w.g3m\It_Configures---A34EAFFF\HelloWorld.csproj]
        
            0 Warning(s)
            7 Error(s)
```

Reply:
ouch. Can you open an issue for this?

Reply:
You should be able to use `[RequiresMSBuildVersionFact("17.0.0.32901")]` for this, and it will skip the test when running on the older MSBuild, but automatically start to run again on Full Framework once CI is updated to a newer VS.

@pranavkm @eerhardt